src/main/java/com/google/cloud/genomics/dataflow/pipelines/VerifyBamId.java

```diff
@@ -0,0 +1,486 @@
+import com.google.cloud.genomics.dataflow.readers.ReadGroupStreamer;
+import com.google.cloud.genomics.dataflow.readers.VariantStreamer;
+import com.google.cloud.genomics.dataflow.utils.GCSOptions;
+import com.google.cloud.genomics.dataflow.utils.GenomicsDatasetOptions;
+import com.google.cloud.genomics.dataflow.utils.GenomicsOptions;
+import com.google.cloud.genomics.dataflow.utils.ReadFunctions;
+import com.google.cloud.genomics.dataflow.utils.Solver;
+import com.google.cloud.genomics.dataflow.utils.VariantFunctions;
+import com.google.cloud.genomics.utils.GenomicsFactory;
+import com.google.cloud.genomics.utils.GenomicsUtils;
+import com.google.cloud.genomics.utils.ShardBoundary;
+import com.google.cloud.genomics.utils.ShardUtils;
+import com.google.cloud.genomics.utils.ShardUtils.SexChromosomeFilter;
+import com.google.common.collect.ImmutableMultiset;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multiset;
+import com.google.genomics.v1.Read;
+import com.google.genomics.v1.StreamVariantsRequest;
+import com.google.genomics.v1.Variant;
+import com.google.protobuf.ListValue;
+
+/**
+ * Test a set of reads for contamination.
+ *
+ * Takes a set of specified ReadGroupSets of reads to test and statistics on reference allele
+ * frequencies for SNPs with a single alternative from a specified set of VariantSets.
+ * 
+ * See http://googlegenomics.readthedocs.org/en/latest/use_cases/perform_quality_control_checks/verify_bam_id.html
+ * for running instructions.
+ *
+ * Uses the sequence data alone approach described in:
+ * G. Jun, M. Flickinger, K. N. Hetrick, Kurt, J. M. Romm, K. F. Doheny,
+ * G. Abecasis, M. Boehnke,and H. M. Kang, Detecting and Estimating
+ * Contamination of Human DNA Samples in Sequencing and Array-Based Genotype
+ * Data, American journal of human genetics doi:10.1016/j.ajhg.2012.09.004
+ * (volume 91 issue 5 pp.839 - 848)
+ * http://www.sciencedirect.com/science/article/pii/S0002929712004788
+ */
+public class VerifyBamId {
+
+  private static VerifyBamId.VerifyBamIdOptions options;
+  private static Pipeline p;
+  private static GenomicsFactory.OfflineAuth auth;
+  
+  /**
+   * String prefix used for sampling hash function
+   */
+  private static final String HASH_PREFIX = "";
+  
+  // TODO: this value is not quite correct. Test again after
+  // https://github.com/googlegenomics/utils-java/issues/48
+  private static final String VARIANT_FIELDS = "variants(start,calls(genotype,callSetName))";
+
+  /**
+   * Options required to run this pipeline.
+   */
+  public static interface VerifyBamIdOptions extends GenomicsDatasetOptions, GCSOptions {
+
+    @Description("A comma delimited list of the IDs of the Google Genomics ReadGroupSets this "
+        + "pipeline is working with. Default (empty) indicates all ReadGroupSets in InputDatasetId."
+        + "  This(and variantSetIds) or InputDatasetId must be set.  InputDatasetId overrides "
```

Comment:
Updated.  It also needed a correction since the variantSet from which we get the allelic frequencies will generally _not_ be in the same dataset as the reads.